src/interceptors/response.interceptor.ts

```diff
@@ -0,0 +1,46 @@
+import {
+  Injectable,
+  NestInterceptor,
+  ExecutionContext,
+  CallHandler,
+  Logger,
+} from '@nestjs/common';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+
+export interface Response<T> {
+  data: T;
+}
+
+@Injectable()
+export class ResponseInterceptor<T> implements NestInterceptor<T, Response<T>> {
+  private logger = new Logger();
+
+  intercept(
+    context: ExecutionContext,
+    next: CallHandler,
+  ): Observable<Response<T>> {
+    const statusCode = context.switchToHttp().getResponse().statusCode;
+    const req = context.switchToHttp().getRequest();
+    const userDetail = req?.user || {};
+
```

Comment:
PullProofAI

The `ResponseInterceptor` class seems well-implemented. However, there are a few areas that could be improved for better readability and maintainability:

1. Consider extracting the logic inside the `map` function into separate methods for better separation of concerns.
2. Use destructuring to simplify accessing properties like `data.message`.
3. Ensure consistent use of ternary operators for conditions.

Here's a suggested refactoring for the `map` function inside the `intercept` method:

```diff
+ const formatData = (data) => {
+   return data && data.message ? { ...data, message: undefined } : data;
+ };
+
+ const response = {
+   isSuccess: statusCode >= 200 && statusCode < 300,
+   statusCode,
+   error: formatData(data).message,
+   message: formatData(data).message,
+   data: formatData(data),
+ };
```

Overall, the code looks good, but these improvements can enhance its readability and maintainability.



<!-- This is an auto-generated comment by PullProofAI -->